internal/resolver/dns/dns_resolver_test.go

```diff
@@ -1490,55 +1498,63 @@ func TestRateLimitedResolve(t *testing.T) {
 		t.Fatalf("delegate resolver returned unexpected type: %T\n", tr)
 	}
 
-	// Observe the time before unblocking the lookupHost call.  The 100ms rate
-	// limiting timer will begin immediately after that.  This means the next
-	// resolution could happen less than 100ms if we read the time *after*
-	// receiving from tr.ch
-	start := time.Now()
+	ctx, ctxCancel := context.WithTimeout(context.Background(), defaultTestTimeout)
```

Comment:
Naming nit: `ctxCancel`->`cancel` is the preferred name.

Reply:
Ah sounds good, switched.